From c79aa44d58ef29e22516986641970c343cb15cd0 Mon Sep 17 00:00:00 2001
From: Petr Tesarik <ptesarik@suse.com>
Date: Tue, 25 Feb 2020 14:47:02 +0100
Subject: Make sure that the udev runtime directory exists
References: bsc#1164713
Upstream: merged
Git-commit: c79aa44d58ef29e22516986641970c343cb15cd0

Hotplug operations will not work with kdump if the udev runtime
directory does not exist at startup.

While fixing it, move udev rules handling to load.sh / unload.sh.

Signed-off-by: Petr Tesarik <ptesarik@suse.com>
---
 init/kdump.service |    2 --
 init/load.sh       |    3 +++
 init/unload.sh     |    2 ++
 3 files changed, 5 insertions(+), 2 deletions(-)

--- a/init/kdump.service
+++ b/init/kdump.service
@@ -5,8 +5,6 @@ After=local-fs.target network.service Ya
 [Service]
 Type=oneshot
 ExecStart=/lib/kdump/load.sh --update
-ExecStartPost=-/usr/bin/cp /usr/lib/kdump/70-kdump.rules /run/udev/rules.d/70-kdump.rules
-ExecStopPost=-/usr/bin/rm -f /run/udev/rules.d/70-kdump.rules
 ExecStop=/lib/kdump/unload.sh
 RemainAfterExit=true
 
--- a/init/load.sh
+++ b/init/load.sh
@@ -7,6 +7,7 @@ KDUMPTOOL=/usr/sbin/kdumptool
 KEXEC=/sbin/kexec
 FADUMP_ENABLED=/sys/kernel/fadump_enabled
 FADUMP_REGISTERED=/sys/kernel/fadump_registered
+UDEV_RULES_DIR=/run/udev/rules.d
 
 #
 # Remove an option from the kernel command line
@@ -324,6 +325,8 @@ result=$?
 
 if [ $result = 0 ] ; then
     echo 1 > /proc/sys/kernel/panic_on_oops
+    mkdir -p "$UDEV_RULES_DIR"
+    cp /usr/lib/kdump/70-kdump.rules "$UDEV_RULES_DIR"/70-kdump.rules
 fi
 
 exit $result
--- a/init/unload.sh
+++ b/init/unload.sh
@@ -6,6 +6,7 @@
 KDUMPTOOL=/usr/sbin/kdumptool
 KEXEC=/sbin/kexec
 FADUMP_REGISTERED=/sys/kernel/fadump_registered
+UDEV_RULES_DIR=/run/udev/rules.d
 
 eval $($KDUMPTOOL dump_config)
 
@@ -21,4 +22,5 @@ fi
 
 test $? -eq 0 || exit 1
 
+rm -f "$UDEV_RULES_DIR"/70-kdump.rules
 exit 0
